cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ProgramNotificationSubscriberService.java

```diff
@@ -409,6 +409,16 @@ private RunRecordDetail handleProgramCompletion(AppMetadataStore appMetadataStor
 
       getEmitMetricsRunnable(programRunId, recordedRunRecord,
                              STATUS_METRICS_NAME.get(programRunStatus)).ifPresent(runnables::add);
+
+      Map<String, String> systemArgs = recordedRunRecord.getSystemArgs();
+      boolean skipProvisioning = Boolean.parseBoolean(systemArgs.get(ProgramOptionConstants.SKIP_PROVISIONING));
+      if (!skipProvisioning) {
+        // emit provisioning time metric if this is not a preview run.
```

Comment:
It seems strange to emit provisioning metrics when program complete. It should be done in place where provisioning was skipped (I believe is in the `handleProgramEvent` method). Also, provisioning can be skipped not just because of preview, so it's better change the comment.

Reply:
Removed unnecessary skipProvisioning check and fixed the comment.